Feature request: privacy statement and imprint

schuettloeffel-elsa opened the pull request:
closes #297

- Added Imprint and Privacy links to footer
- Implemented fetching of imprint and privacy html files
- Translated both terms to all languages
- Added test files
- also see backend PR: https://github.com/opencast/opencast/pull/5674

<img width="1550" alt="image" src="https://github.com/opencast/opencast-admin-interface/assets/53210931/0fca87c3-c011-4d47-b939-83828e786721">

Arnei commented on app/src/components/About.tsx:
```diff
@@ -0,0 +1,61 @@
+				</nav>
+			</section>
+			<div className="about">
+				<div dangerouslySetInnerHTML={{ __html: aboutContent }} ></div>
```
In the case that there is not conent, the user is met with an empty gray block. Having some form of error message ("The imprint is unfortunately unavailable at this time") would be good.

Arnei commented on app/src/styles/components/_components-config.scss:
```diff
@@ -40,6 +40,8 @@
 @import "stats";
 @import "popover";
 @import "collapsible-box";
+@import "collapsible-box";
```
Importing this twice is probably not intended?

schuettloeffel-elsa replied:
Thanks, will remove it.

Arnei reviewed (commented):
Played around with this a bit and what can I say, it works :+1: I'll let others comment on whether the design of the pages is in line with the admin ui design docs, and instead raise some discussions points about technical stuff below.

I would prefer if we could avoid using user-generated HTML and JSON or some other form of string map instead. This would allow us more control in the frontend to ensure everything looks well, users don't have to fiddle around with HTML, and we could get some stuff like "Tel." or "Fax" translated in the frontend so users don't have to.
But I am also not at all knowledgable about the requirements for imprints and privacy statements, so if you say they absolutely require this degree of flexiblity I'm fine with it.

schuettloeffel-elsa commented:
> I would prefer if we could avoid using user-generated HTML and JSON or some other form of string map instead. This would allow us more control in the frontend to ensure everything looks well, users don't have to fiddle around with HTML, and we could get some stuff like "Tel." or "Fax" translated in the frontend so users don't have to. But I am also not at all knowledgable about the requirements for imprints and privacy statements, so if you say they absolutely require this degree of flexiblity I'm fine with it.

In my experience at least the privacy statements need that degree of flexibility since they are highliy unique per organization and would be almost impossible to be generalized into a template of some kind (e.g. some organizations use kind of tables in their privacy statements to visualize data access).

schuettloeffel-elsa commented on app/src/components/About.tsx:
```diff
@@ -0,0 +1,61 @@
+				</nav>
+			</section>
+			<div className="about">
+				<div dangerouslySetInnerHTML={{ __html: aboutContent }} ></div>
```
Hm.. if we use plain html in the config files, I doubt there is an alternative to dangerouslySetInnerHTML(). But sanitizing it is still a good idea, gonna implement it.

Arnei commented:
> > I would prefer if we could avoid using user-generated HTML and JSON or some other form of string map instead. This would allow us more control in the frontend to ensure everything looks well, users don't have to fiddle around with HTML, and we could get some stuff like "Tel." or "Fax" translated in the frontend so users don't have to. But I am also not at all knowledgable about the requirements for imprints and privacy statements, so if you say they absolutely require this degree of flexiblity I'm fine with it.
> 
> In my experience at least the privacy statements need that degree of flexibility since they are highliy unique per organization and would be almost impossible to be generalized into a template of some kind (e.g. some organizations use kind of tables in their privacy statements to visualize data access).

Unfortunate, but oh well, it's just like that sometimes :D. In that case I'm fine with approving this once you get around to the other comments.

gregorydlogan commented on app/src/components/Footer.tsx:
```diff
@@ -19,15 +22,20 @@ const Footer: React.FC = () => {
+							)}
+						</li>
+					)}
+					<li><Link to="/about/imprint">{t("IMPRINT")}</Link></li>
```
I'm wondering if we should hide these by default, or at least provide a way to turn them off?  I'm sure the majority of institutions will (eventually) get to it, but for a lot of early adopters I can see not wanting boilerplate text linked from here.

schuettloeffel-elsa commented:
Sorry for the delay, had lots of other stuff to do, I addressed all your concerns-

@Arnei: can you have a look now? 
@gregorydlogan: I implemented a switch to turn on/off both Footer Elements, you can configure it in `org.opencastproject.admin.display_about` in the me.json file.

schuettloeffel-elsa commented on app/src/i18n/org/opencastproject/adminui/languages/lang-de_DE.json:
```diff
@@ -2139,5 +2139,10 @@
 			"medium": "{{time, HH:mm:ss}}",
 			"full": "{{time, HH:mm [Uhr] z}}"
 		}
+	},
+	"ABOUT": {
+		"IMPRINT": "Impressum",
+		"PRIVACY": "Datenschutzerklärung",
+		"NOCONTENT": "Inhalt nicht verfügbar"
```
is this also true for new variables?

ziegenberg replied:
Yes. Updates to language strings (inserts, updates, deletes) come via Crowdin.

schuettloeffel-elsa replied:
why didnt i knew that before translation to all these languages 😄

schuettloeffel-elsa replied:
removed all non en_us translations now

ziegenberg commented on app/src/components/About.tsx:
```diff
@@ -0,0 +1,63 @@
+						setAboutContent(t("ABOUT.NOCONTENT").toString());
+					});
+			});
+
+	}, [location.pathname]); // Listen to changes in pathname
```
`eslint` complains here:
```
React Hook useEffect has a missing dependency: 't'.
```
This warning is "fixed" this way in other places in the code the same way.

```suggestion
           // eslint-disable-next-line react-hooks/exhaustive-deps
	}, [location.pathname]); // Listen to changes in pathname
```

ziegenberg commented on app/src/components/Footer.tsx:
```diff
@@ -5,12 +5,15 @@ import {
 } from "../selectors/userInfoSelectors";
 import { hasAccess } from "../utils/utils";
 import { useAppSelector } from "../store";
+import { Link, useNavigate } from "react-router-dom";
```
`eslint` complains:
```
'useNavigate' is defined but never used
```
Can this go? Do we really need this?

```suggestion
import { Link } from "react-router-dom";
```

ziegenberg commented:
I did a build run and found some warnings.

schuettloeffel-elsa commented:
Thanks, commitet your suggestions
